yellow-paper/docs/public-vm/state.md

```diff
@@ -0,0 +1,107 @@
+
+This trace of an AVM session's contract calls and world state accesses is named the **world state access trace**.
+
+> The world state access trace is also important for enforcing limitations on the maximum number of allowable world state accesses.
+
+#### _WorldStateAccessTrace_
+
+Each entry in the world state access trace is listed below along with its type and the instructions that append to it:
+
+| Trace                 | Relevant State     | Trace Vector Type                  | Instructions         |
+| ---                   | ---                | ---                                | ---                  |
+| `publicStorageReads`  | `publicDataDB`     | `Vector<TracedStorageRead>`        | `SLOAD`              |
+| `publicStorageWrites` | `publicDataDB`     | `Vector<TracedStorageWrite>`       | `SSTORE`             |
+| `l1ToL2MessageReads`  | `l1ToL2MessagesDB` | `Vector<TracedL1ToL2MessageRead>`  | `READL1TOL2MSG`      |
+| `newNoteHashes`       | `noteHashDB`       | `Vector<TracedNoteHash>`           | `EMITNOTEHASH`       |
+| `newNullifiers`       | `nullifierDB`      | `Vector<TracedNullifier>`          | `EMITNULLIFIER`      |
+| `nullifierChecks`     | `nullifierDB`      | `Vector<TracedIndexedLeafCheck>`   | `CHECKNULLIFIER`     |
+| `archiveChecks`       | `archiveDB`        | `Vector<TracedLeafCheck>`          | `CHECKARCHIVE`       |
+| `<tree>RootReads`     | `<tree>DB`         | `Vector<TracedTreeRootRead>`       | `<TREE>ROOT`         |
+| `contractCalls`       | `contract*DB`      | `Vector<AztecAddress>`             | `*CALL`              |
+
+> The types tracked in these trace vectors are defined [here](./type-structs).
+
+> The syntax `*CALL` is short for `CALL`/`STATICCALL`/`DELEGATECALL`.
+
+> There is a "root read" trace and instruction for each tree (`noteHashRootReads`, `NOTEHASHROOT`, `nullifierRootReads`, `NULLIFIERROOT`, etc.).
+
+
+A world state tree may be accessed in one or more ways from within the AVM. For a given tree, each supported world state access type has a dedicated instruction and access trace. This information is presented in the table below.
+
+| Instructions        | Description |
+| ---                 | ---         |
+| `SLOAD`             | Read a public storage slot value. Trigger a membership check of its leaf in the public data tree. |
+| `SSTORE`            | Write a value to a public storage slot. Trigger an update to its leaf in the public data tree.    |
+| `READL1TOL2MSG`     | Read an entire L1-to-L2 message preimage and trigger a membership check of its message tree leaf. |
+| `EMITNOTEHASH`      | Append a note hash to the note hash tree. |
+| `EMITNULLIFIER`     | Append a nullifier to the nullifier tree. |
+| `CHECKNULLIFIER`    | Trigger a membership check of the specified leaf of the nullifier tree. |
```

Comment:
oh this is new!

Reply:
Yeah I'm trying to get the YP up to date with our latest hopes for tree-access operations. Will get them in here and then pass over to Mike and Alvaro to see what we're missing.